### ISSUE TITLE
Data race on `main` branch

### ISSUE THREAD

**Issue by tzdybal** [rollkit/rollkit#264]
https://github.com/celestiaorg/optimint/runs/4957750474?check_suite_focus=true#step:5:2789

```WARNING: DATA RACE
Read at 0x00c000612d43 by goroutine 282:
  testing.(*common).logDepth()
      /opt/hostedtoolcache/go/1.17.6/x64/src/testing/testing.go:768 +0xc4
  testing.(*common).log()
      /opt/hostedtoolcache/go/1.17.6/x64/src/testing/testing.go:761 +0x5a
  testing.(*common).Log()
      /opt/hostedtoolcache/go/1.17.6/x64/src/testing/testing.go:800 +0x14
  github.com/celestiaorg/optimint/log/test.(*TestLogger).Error()
      /home/runner/work/optimint/optimint/log/test/loggers.go:33 +0x210
  github.com/celestiaorg/optimint/p2p.(*Gossiper).ProcessMessages()
      /home/runner/work/optimint/optimint/p2p/gossip.go:95 +0x199
  github.com/celestiaorg/optimint/p2p.(*Client).setupGossiping·dwrap·2()
      /home/runner/work/optimint/optimint/p2p/client.go:309 +0x58

Previous write at 0x00c000612d43 by goroutine 28:
  testing.tRunner.func1()
      /opt/hostedtoolcache/go/1.17.6/x64/src/testing/testing.go:1246 +0x584
  testing.tRunner()
      /opt/hostedtoolcache/go/1.17.6/x64/src/testing/testing.go:1265 +0x268
  testing.(*T).Run·dwrap·21()
      /opt/hostedtoolcache/go/1.17.6/x64/src/testing/testing.go:1306 +0x47

Goroutine 282 (running) created at:
  github.com/celestiaorg/optimint/p2p.(*Client).setupGossiping()
      /home/runner/work/optimint/optimint/p2p/client.go:309 +0x329
  github.com/celestiaorg/optimint/p2p.(*Client).startWithHost()
      /home/runner/work/optimint/optimint/p2p/client.go:113 +0x17e
  github.com/celestiaorg/optimint/p2p.startTestNetwork()
      /home/runner/work/optimint/optimint/p2p/utils_test.go:117 +0x9da
  github.com/celestiaorg/optimint/p2p.TestGossiping()
      /home/runner/work/optimint/optimint/p2p/client_test.go:105 +0x9d4
  testing.tRunner()
      /opt/hostedtoolcache/go/1.17.6/x64/src/testing/testing.go:1259 +0x22f
  testing.(*T).Run·dwrap·21()
      /opt/hostedtoolcache/go/1.17.6/x64/src/testing/testing.go:1306 +0x47

Goroutine 28 (running) created at:
  testing.(*T).Run()
      /opt/hostedtoolcache/go/1.17.6/x64/src/testing/testing.go:1306 +0x726
  testing.runTests.func1()
      /opt/hostedtoolcache/go/1.17.6/x64/src/testing/testing.go:1598 +0x99
  testing.tRunner()
      /opt/hostedtoolcache/go/1.17.6/x64/src/testing/testing.go:1259 +0x22f
  testing.runTests()
      /opt/hostedtoolcache/go/1.17.6/x64/src/testing/testing.go:1596 +0x7ca
  testing.(*M).Run()
      /opt/hostedtoolcache/go/1.17.6/x64/src/testing/testing.go:1504 +0x9d1
  main.main()
      _testmain.go:127 +0x32b
```

**Comment by tzdybal** [comment]
Looks like invalid use of test logger.

**Comment by nashqueue** [comment]
Fixed by: #916
